File: ruby/grains/grains.rb
class Grains
  TOTAL_SQUARES = 64

  def self.square(n)
    raise ArgumentError, "Must be between 1 and #{TOTAL_SQUARES}" if n < 1 || n > 64
    2 ** (n - 1)
  end

  def self.total
    (1..TOTAL_SQUARES).reduce(0) { |total, n| total += self.square(n) }
  end
end

File: javascript/grains/grains.js
var bigInt = require('./big-integer');

var Grains = function() {
  this.squares = 64;
};

Grains.prototype.square = function(num) {
  if (num < 1 || num > 64) { throw 'InvalidNumber' };
  return bigInt(2).pow(num - 1).toString();
};

Grains.prototype.total = function() {
  var total = bigInt(0);
  for (var i = 1; i <= this.squares; i++) {
    total = total.add(this.square(i));
  }
  return total.toString();
}

module.exports = Grains;

var bigInt = require('./big-integer');

var Grains = function() {
  this.squares = 64;
};
